modules/apps/headless/headless-admin-user/headless-admin-user-impl/src/main/java/com/liferay/headless/admin/user/internal/resource/v1_0/EmailAddressResourceImpl.java

```diff
@@ -201,27 +205,68 @@ private Long _getListTypeId(String className, String name) {
 		return listType.getListTypeId();
 	}
 
+	private boolean _hasPrimary(
+		List<com.liferay.portal.kernel.model.EmailAddress> emailAddresses) {
+
+		for (com.liferay.portal.kernel.model.EmailAddress emailAddress :
+				emailAddresses) {
+
+			if (emailAddress.isPrimary()) {
+				return true;
+			}
+		}
+
+		return false;
+	}
+
 	private EmailAddress _updateEmailAddress(
 			EmailAddress emailAddress,
-			com.liferay.portal.kernel.model.EmailAddress serviceEmailAddress)
+			com.liferay.portal.kernel.model.EmailAddress
+				serviceBuilderEmailAddress)
 		throws Exception {
 
-		serviceEmailAddress = _emailAddressService.updateEmailAddress(
+		serviceBuilderEmailAddress = _emailAddressService.updateEmailAddress(
 			GetterUtil.getString(
 				emailAddress.getExternalReferenceCode(),
-				serviceEmailAddress.getExternalReferenceCode()),
-			serviceEmailAddress.getEmailAddressId(),
+				serviceBuilderEmailAddress.getExternalReferenceCode()),
+			serviceBuilderEmailAddress.getEmailAddressId(),
 			GetterUtil.getString(
 				emailAddress.getEmailAddress(),
-				serviceEmailAddress.getAddress()),
+				serviceBuilderEmailAddress.getAddress()),
 			GetterUtil.getLong(
 				_getListTypeId(
-					serviceEmailAddress.getClassName(), emailAddress.getType()),
-				serviceEmailAddress.getListTypeId()),
+					serviceBuilderEmailAddress.getClassName(),
+					emailAddress.getType()),
+				serviceBuilderEmailAddress.getListTypeId()),
 			GetterUtil.getBoolean(
-				emailAddress.getPrimary(), serviceEmailAddress.isPrimary()));
+				emailAddress.getPrimary(),
+				serviceBuilderEmailAddress.isPrimary()));
+
+		List<com.liferay.portal.kernel.model.EmailAddress> emailAddresses =
+			_emailAddressService.getEmailAddresses(
+				serviceBuilderEmailAddress.getClassName(),
+				serviceBuilderEmailAddress.getClassPK());
+
+		if (!_hasPrimary(emailAddresses)) {
+			long size = emailAddresses.size();
+
+			for (com.liferay.portal.kernel.model.EmailAddress tempEmailAddress :
+					emailAddresses) {
+
+				if ((size == 1) ||
+					(tempEmailAddress.getEmailAddressId() !=
+						serviceBuilderEmailAddress.getEmailAddressId())) {
+
+					_emailAddressService.updateEmailAddress(
+						tempEmailAddress.getExternalReferenceCode(),
+						tempEmailAddress.getEmailAddressId(),
+						tempEmailAddress.getAddress(),
+						tempEmailAddress.getListTypeId(), true);
+				}
+			}
+		}
```

Comment:
this should be done only if "previous isPrimary" was true e now false right?
I mean.. if it was false and now true should already throw an error
if it was false and now false ... it's already fine
if it was true and now true ... it's already fine

Reply:
and then loop
--- if size == 1 || email != email then update and break; (otherwise it will continue the loop)

Reply:
I don't think we need to know if there are other with primary (so the _hasPrimary method I think is not useful)